services/visual_recognition/v3.js

```diff
@@ -62,301 +37,331 @@ module.exports = function (RED) {
+  function addTask (asyncTasks, msg, key, listParams, node) {
+    asyncTasks.push(function (callback) {
+      var buffer = msg.params[key];
+      temp.open({suffix: '.' + fileType(buffer).ext}, function (err, info) {
+        if (err) {
           node.status({fill:'red', shape:'ring', 
-                       text:'call to watson visual recognition v3 service failed'}); 
-          msg.result = {};
-          msg.result['error_code']= err.code;
-          if (!err.error)
-            msg.result['error']= err.error;
-          console.log('Error:', err.code);
-          node.error('Error code : ' +  err.code);
-          return;
-        }
-        else if (body.images[0].error)
-        {
-          var err_desc = body.images[0].error.description;
-          var err_id = body.images[0].error.error_id;
-          node.status({fill:'red', shape:'ring', 
-                       text:'call to watson visual recognition v3 service failed'}); 
-          msg.result = {};
-          msg.result['error_id']= err_id;
-          msg.result['error']= err_desc;
-          console.log('Error:', err_desc);
-          msg.payload='see msg.result.error';
-          node.send(msg); 
-        }
-        else {
-          msg.result = {};
-          msg.result['all'] = body;
-          msg.payload='see msg.result'; // to remove any Buffer that could remains
-          node.send(msg); 
-          node.status({});
-        }
-      }; // actionComplete    
-
-      var actionComplete2 = function(err, body, other) {
+                       text:'unable to open image stream'});          
+          node.error('Node has been unable to open the image stream', msg);
+          return callback('open error on '+key);
+        }  
+        stream_buffer(info.path, msg.params[key], function () {
+          listParams[key]=fs.createReadStream(info.path);
+          callback(null, key);
+        });
+      });
+    });
+  }
 
-        if (err != null && body==null)
-        {
-          node.status({fill:'red', shape:'ring', text:'call to watson visual recognition v3 service failed'}); 
-          msg.result = {};
-          msg.result['error_code']= err.code;
-          if (!err.error)
-            msg.result['error']= err.error;
-          console.log('Error:', err.code);
-          node.error('Error code : ' +  err.code);
-          return;
-        }
-        else {
-          msg.result = {};
-          msg.result['all'] = body;
-          msg.payload='see msg.result'; // to remove any Buffer that could remains
-          node.send(msg); 
-          node.status({});
+  function prepareParamsCreateClassifier (params, node, msg, cb) {
+    var listParams = {}, asyncTasks = [] , key=null;
+    for (key in msg.params) {
+      if (key.indexOf('_examples')>=0) {
+        addTask(asyncTasks, msg, key, listParams, node);
+      } else if (key==='name') {
+        listParams[key]=msg.params[key];
+      }
+    } // for
+    async.parallel(asyncTasks, function(error, results){
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['results' is defined but never used.](https://www.codacy.com/app/node-red-nodes-watson/node-red-node-watson/file/2737047197/issues/source?bid=3225115&fileBranchId=3352493#l205)